helix-lsp/src/client.rs

```diff
@@ -861,6 +861,18 @@ impl Client {
         position: lsp::Position,
         new_name: String,
     ) -> anyhow::Result<lsp::WorkspaceEdit> {
+        let capabilities = self.capabilities.get().unwrap();
+
+        // check if we're able to rename
+        match capabilities.rename_provider {
+            Some(lsp::OneOf::Left(true)) | Some(lsp::OneOf::Right(_)) => (),
+            // None | Some(false)
+            _ => {
+                log::warn!("the server does not support rename");
+                return Ok(lsp::WorkspaceEdit::default());
```

Comment:
It would be better to show an editor error with `editor.set_error` in the calling code if it returns an `Err` variant from here.

Reply:
thanks point out, after pushed, I also found this and updating now

Reply:
tested ok:

![image](https://user-images.githubusercontent.com/41882455/164306658-1691ede5-9a73-4a2b-9c54-0e0da89c4dc3.png)

Reply:
code updated, PTAL